### ISSUE TITLE
systemd-boot: Allow configuration using SMBIOS type 11 OEM strings 

### ISSUE THREAD

**Issue by daandemeyer** [systemd/systemd#26066]
### Is your feature request related to a problem? Please describe

We already read systemd credentials from the SMBIOS type 11 table in userspace. Let's extend systemd-boot to read credentials from this table as well to allow configuring systemd-boot using credentials. A first usecase I can think of is a credential `kernel.extra` or so to pass additional kernel command line arguments to systemd-boot which will forward them to the kernel.

**Comment by poettering** [comment]
Let's not mix up "SMBIOS type 11 variables" with "credentials". The latter is a systemd thing and can be propagated, encrypted and so on. I think sd-boot should read certain SMBIOS type 11 variables, but i am not sure it should bother with the "credentials" concept, because that would require TPM stuff and so on, and there's no point in that.

or in other words, I am pretty sure sd-boot should react to a qemu switch like this:

```
        -smbios type=11,value=io.systemd.boot.kernel-extra=debug \
```

but not to one like this:

```
        -smbios type=11,value=io.systemd.credential:boot.kernel-extra=debug \
```

if you follow what I am trying to say.

And yeah, I'd love such a feature.

**Comment by poettering** [comment]
(in fact, there's a TODO list item for this already)

**Comment by poettering** [comment]
https://github.com/systemd/systemd/pull/26005 already contains half an smbios parser btw, so building on top of that it should be easy to implement this RFE here too.

**Comment by medhefgo** [comment]
Fixed with #26258
